walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -96,19 +102,57 @@
     return result;
   }
 
+ /**
+  * Given a list of unavailable timeranges and a meeting request, this method 
+  * finds available timeranges to meet in .
+  *
+  * @param  unavailableTimes  A list of unavalable timeranges.
+  * @param  request  A meeting request.
+  * @return List of timeranges when a meeting could happen.
+  */
+  public ArrayList<TimeRange> findAvailableTimes(ArrayList<TimeRange> unavailableTimes, MeetingRequest request) {
+    ArrayList<TimeRange> availableTimes = new ArrayList<>();
+    TimeRange possibleTime;
+    int prevEndTime = TimeRange.START_OF_DAY;
+    
+    for (TimeRange curTime: unavailableTimes){
+
+      possibleTime = TimeRange.fromStartEnd(prevEndTime, curTime.start(), false);
+
+      if (possibleTime.duration() >= request.getDuration()){
+        availableTimes.add(possibleTime);
```

Comment:
So if I'm reading this right.  For every single unavailable time we are adding a single available time.

This algorithm makes the assumption that all unavailable times are sorted and do not overlap right?

Reply:
That's right. I just edited the method description to explicitly state that.